Subject: [PATCH] [ANSIENG-4166] | added jq support
[ANSIENG-4166] | added vim and jq packages
---
Index: molecule/Dockerfile-amazonlinux2023.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-amazonlinux2023.j2 b/molecule/Dockerfile-amazonlinux2023.j2
--- a/molecule/Dockerfile-amazonlinux2023.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-amazonlinux2023.j2	(revision 8481a608f5ca6ef4c512dd6399f62261f1210fcf)
@@ -6,6 +6,7 @@
 RUN yum -y install rpm \
  && yum -y update \
  && yum -y install sudo vim-enhanced \
+ && yum install -y jq \
  && yum -y install tar procps-ng\
  && yum install -y systemd \
  && yum clean all
Index: molecule/Dockerfile-debian-archive.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian-archive.j2 b/molecule/Dockerfile-debian-archive.j2
--- a/molecule/Dockerfile-debian-archive.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian-archive.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -24,7 +24,9 @@
       ca-certificates \
       openssl \
       unzip \
-      curl
+      curl \
+      vim \
+      jq
 
 # Hack to fix cacerts on debian hosts, preventing confluenthub installs from working
 RUN mkdir -p /etc/ssl/certs/java/
Index: molecule/Dockerfile-debian-java8.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian-java8.j2 b/molecule/Dockerfile-debian-java8.j2
--- a/molecule/Dockerfile-debian-java8.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian-java8.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -18,7 +18,9 @@
       ca-certificates \
       openssl \
       unzip \
-      curl
+      curl \
+      vim \
+      jq
 
 # Hack to fix cacerts on debian hosts, preventing confluenthub installs from working
 RUN /usr/bin/printf '\xfe\xed\xfe\xed\x00\x00\x00\x02\x00\x00\x00\x00\xe2\x68\x6e\x45\xfb\x43\xdf\xa4\xd9\x92\xdd\x41\xce\xb6\xb2\x1c\x63\x30\xd7\x92' > /etc/ssl/certs/java/cacerts && \
Index: molecule/Dockerfile-debian.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian.j2 b/molecule/Dockerfile-debian.j2
--- a/molecule/Dockerfile-debian.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -23,7 +23,9 @@
       ca-certificates \
       openssl \
       unzip \
-      curl
+      curl \
+      vim \
+      jq
 
 # Hack to fix cacerts on debian hosts, preventing confluenthub installs from working
 RUN mkdir -p /etc/ssl/certs/java/
Index: molecule/Dockerfile-debian10-archive.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian10-archive.j2 b/molecule/Dockerfile-debian10-archive.j2
--- a/molecule/Dockerfile-debian10-archive.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian10-archive.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -19,7 +19,9 @@
       ca-certificates \
       openssl \
       unzip \
-      curl
+      curl \
+      vim \
+      jq
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
 {% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
Index: molecule/Dockerfile-debian10-java11.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian10-java11.j2 b/molecule/Dockerfile-debian10-java11.j2
--- a/molecule/Dockerfile-debian10-java11.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian10-java11.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -11,7 +11,9 @@
       ca-certificates \
       openssl \
       unzip \
-      curl
+      curl \
+      vim \
+      jq
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
 {% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
Index: molecule/Dockerfile-debian10.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian10.j2 b/molecule/Dockerfile-debian10.j2
--- a/molecule/Dockerfile-debian10.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian10.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -19,7 +19,9 @@
       ca-certificates \
       openssl \
       unzip \
-      curl
+      curl \
+      vim \
+      jq
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
 {% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
Index: molecule/Dockerfile-debian12-java17.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian12-java17.j2 b/molecule/Dockerfile-debian12-java17.j2
--- a/molecule/Dockerfile-debian12-java17.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian12-java17.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -16,7 +16,9 @@
       sudo \
       wget \
       tar \
-      systemd systemd-sysv dbus
+      systemd systemd-sysv dbus \
+      vim \
+      jq
 
 # Create a systemd service for running in the container
 RUN mkdir -p /etc/systemd/system/docker.service.d && \
Index: molecule/Dockerfile-debian12.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-debian12.j2 b/molecule/Dockerfile-debian12.j2
--- a/molecule/Dockerfile-debian12.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-debian12.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -15,7 +15,9 @@
       sudo \
       wget \
       tar \
-      systemd systemd-sysv dbus
+      systemd systemd-sysv dbus \
+      vim \
+      jq
 
 {% set ARCHITECTURE_SHORTHAND = lookup('env', 'ARCHITECTURE_SHORTHAND') | default ('x64', true) %}
 
Index: molecule/Dockerfile-ubuntu-archive.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-ubuntu-archive.j2 b/molecule/Dockerfile-ubuntu-archive.j2
--- a/molecule/Dockerfile-ubuntu-archive.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-ubuntu-archive.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -4,5 +4,5 @@
     apt-get update -y && \
     apt-get install -y software-properties-common && \
     apt-get install -y ca-certificates-java && \
-    apt-get install -y openjdk-17-jdk wget gnupg && \
+    apt-get install -y openjdk-17-jdk wget gnupg vim jq && \
     apt-get install -y dirmngr
Index: molecule/Dockerfile-ubuntu-java11.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-ubuntu-java11.j2 b/molecule/Dockerfile-ubuntu-java11.j2
--- a/molecule/Dockerfile-ubuntu-java11.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-ubuntu-java11.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -2,7 +2,7 @@
 
 RUN mkdir -p /usr/share/man/man1 && \
     apt-get update -y && \
-    apt-get install -y openjdk-11-jdk wget gnupg
+    apt-get install -y openjdk-11-jdk wget gnupg vim jq
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
 {% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
Index: molecule/Dockerfile-ubuntu-java8.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-ubuntu-java8.j2 b/molecule/Dockerfile-ubuntu-java8.j2
--- a/molecule/Dockerfile-ubuntu-java8.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-ubuntu-java8.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -2,7 +2,7 @@
 
 RUN mkdir -p /usr/share/man/man1 && \
     apt-get update -y && \
-    apt-get install -y openjdk-8-jdk wget gnupg
+    apt-get install -y openjdk-8-jdk wget gnupg vim jq
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
 {% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
Index: molecule/Dockerfile-ubuntu.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-ubuntu.j2 b/molecule/Dockerfile-ubuntu.j2
--- a/molecule/Dockerfile-ubuntu.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-ubuntu.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -4,7 +4,7 @@
     apt-get update -y && \
     apt-get install -y software-properties-common && \
     apt-get install -y ca-certificates-java && \
-    apt-get install -y openjdk-17-jdk wget gnupg
+    apt-get install -y openjdk-17-jdk wget gnupg vim jq
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
 {% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
Index: molecule/Dockerfile-ubuntu2004-archive.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-ubuntu2004-archive.j2 b/molecule/Dockerfile-ubuntu2004-archive.j2
--- a/molecule/Dockerfile-ubuntu2004-archive.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-ubuntu2004-archive.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -2,5 +2,5 @@
 
 RUN mkdir -p /usr/share/man/man1 && \
     apt-get update -y && \
-    apt-get install -y openjdk-17-jdk wget gnupg && \
+    apt-get install -y openjdk-17-jdk wget gnupg vim jq && \
     apt-get install -y dirmngr
\ No newline at end of file
Index: molecule/Dockerfile-ubuntu2004-java11.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-ubuntu2004-java11.j2 b/molecule/Dockerfile-ubuntu2004-java11.j2
--- a/molecule/Dockerfile-ubuntu2004-java11.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ b/molecule/Dockerfile-ubuntu2004-java11.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
@@ -2,7 +2,7 @@
 
 RUN mkdir -p /usr/share/man/man1 && \
     apt-get update -y && \
-    apt-get install -y openjdk-11-jdk wget gnupg && \
+    apt-get install -y openjdk-11-jdk wget gnupg vim jq && \
     apt-get install -y dirmngr
 
 {% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
Index: molecule/Dockerfile-centos8-base.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-centos8-base.j2 b/molecule/Dockerfile-centos8-base.j2
--- a/molecule/Dockerfile-centos8-base.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
+++ b/molecule/Dockerfile-centos8-base.j2	(revision 8481a608f5ca6ef4c512dd6399f62261f1210fcf)
@@ -26,6 +26,7 @@
 RUN yum -y install rpm \
  && yum -y update \
  && yum -y install sudo vim-enhanced \
+ && yum install -y jq \
  && yum clean all
 
 # Disable requiretty.
Index: molecule/Dockerfile-rhel-java11.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-rhel-java11.j2 b/molecule/Dockerfile-rhel-java11.j2
--- a/molecule/Dockerfile-rhel-java11.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
+++ b/molecule/Dockerfile-rhel-java11.j2	(revision 8481a608f5ca6ef4c512dd6399f62261f1210fcf)
@@ -18,6 +18,7 @@
 RUN yum -y install rpm \
  && yum -y update \
  && yum -y install sudo vim-enhanced \
+ && yum install -y jq \
  && yum clean all
 
 # Disable requiretty.
Index: molecule/Dockerfile-rhel-java17.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-rhel-java17.j2 b/molecule/Dockerfile-rhel-java17.j2
--- a/molecule/Dockerfile-rhel-java17.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
+++ b/molecule/Dockerfile-rhel-java17.j2	(revision 8481a608f5ca6ef4c512dd6399f62261f1210fcf)
@@ -18,6 +18,7 @@
 RUN yum -y install rpm \
  && yum -y update \
  && yum -y install sudo vim-enhanced \
+ && yum install -y jq \
  && yum clean all
 
 # Disable requiretty.
Index: molecule/Dockerfile-rhel-java8.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-rhel-java8.j2 b/molecule/Dockerfile-rhel-java8.j2
--- a/molecule/Dockerfile-rhel-java8.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
+++ b/molecule/Dockerfile-rhel-java8.j2	(revision 8481a608f5ca6ef4c512dd6399f62261f1210fcf)
@@ -18,6 +18,7 @@
 RUN yum -y install rpm \
  && yum -y update \
  && yum -y install sudo vim-enhanced \
+ && yum install -y jq \
  && yum clean all
 
 # Disable requiretty.
Index: molecule/Dockerfile-rhel-tar.j2
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/molecule/Dockerfile-rhel-tar.j2 b/molecule/Dockerfile-rhel-tar.j2
--- a/molecule/Dockerfile-rhel-tar.j2	(revision 7fffaf078517bdd2d9a8ef6fdbd8e765dc48275b)
+++ b/molecule/Dockerfile-rhel-tar.j2	(revision 8481a608f5ca6ef4c512dd6399f62261f1210fcf)
@@ -18,6 +18,7 @@
 RUN yum -y install rpm \
  && yum -y update \
  && yum -y install sudo vim-enhanced \
+&& yum install -y jq \
  && yum clean all
 
 # Disable requiretty.
Index: molecule/Dockerfile-rhel7-java11.j2
===================================================================
diff --git a/molecule/Dockerfile-rhel7-java11.j2 b/molecule/Dockerfile-rhel7-java11.j2
deleted file mode 100644
--- a/molecule/Dockerfile-rhel7-java11.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ /dev/null	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
@@ -1,55 +0,0 @@
-FROM {{ item.image }}
-
-RUN yum -y install java-11-openjdk \
-      rsync \
-      openssl \
-      rsyslog \
-      python3-libselinux \
-      openldap \
-      openldap-servers \
-      openldap-clients \
-      openldap-devel \
-      nss-pam-ldapd \
-      krb5-libs \
-      krb5-server \
-      krb5-workstation \
-      unzip
-
-{% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
-{% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
-{% set REPO_VER = PACKAGE_VER | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') %}
-{% set COMMON_REPO_URL = lookup('env', 'COMMON_REPO_URL') | default('https://packages.confluent.io', true) %}
-{% set CLIENT_REPO_URL = lookup('env', 'CLIENT_REPO_URL') | default('https://packages.confluent.io', true) %}
-
-RUN echo $'[Confluent.clients]\n\
-name=Confluent repository (clients)\n\
-baseurl={{ CLIENT_REPO_URL }}/clients/rpm/centos/\$releasever/\$basearch\n\
-gpgcheck=1\n\
-gpgkey={{ CLIENT_REPO_URL }}/clients/rpm/archive.key\n\
-enabled=1\n\
-\n\
-[Confluent]\n\
-name=Confluent repository\n\
-baseurl={{ COMMON_REPO_URL }}/rpm/{{ REPO_VER }}\n\
-gpgcheck=1\n\
-gpgkey={{ COMMON_REPO_URL }}/rpm/{{ REPO_VER }}/archive.key\n\
-enabled=1' >> /etc/yum.repos.d/confluent.repo
-
-RUN yum clean all && \
-    yum -y install confluent-common \
-      confluent-hub-client \
-      confluent-rebalancer \
-      confluent-rest-utils \
-      confluent-metadata-service \
-      confluent-server \
-      confluent-security \
-      confluent-kafka-connect-replicator \
-      confluent-kafka-rest \
-      confluent-ksqldb \
-      confluent-schema-registry \
-      confluent-control-center-fe \
-      confluent-control-center \
-      confluent-schema-registry-plugins \
-      confluent-ce-kafka-https-server \
-      confluent-ce-kafka-rest \
-      confluent-hub-client
Index: molecule/Dockerfile-rhel7-java17.j2
===================================================================
diff --git a/molecule/Dockerfile-rhel7-java17.j2 b/molecule/Dockerfile-rhel7-java17.j2
deleted file mode 100644
--- a/molecule/Dockerfile-rhel7-java17.j2	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
+++ /dev/null	(revision edad5257e14a7c3593c48e05df1383890d9dae82)
@@ -1,18 +0,0 @@
-FROM {{ item.image }}
-
-# Download Java using wget
-
-RUN yum -y install wget curl
-RUN wget https://download.java.net/java/GA/jdk17.0.2/dfd4a8d0985749f896bed50d7138ee7f/8/GPL/openjdk-17.0.2_linux-x64_bin.tar.gz
-RUN tar xvf openjdk-17.0.2_linux-x64_bin.tar.gz
-RUN sudo mv jdk-17.0.2/ /opt/jdk17/
-RUN rm -rf openjdk-17.0.2_linux-x64_bin.tar.gz
-
-RUN yum -y install rsync \
-      openssl
-
-{% set DEFAULT_PACKAGE_VER = lookup('pipe', "awk '/confluent_package_version:/ {print $2}' $MOLECULE_PROJECT_DIRECTORY/roles/variables/defaults/main.yml" ) %}
-{% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
-{% set REPO_VER = PACKAGE_VER | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') %}
-{% set COMMON_REPO_URL = lookup('env', 'COMMON_REPO_URL') | default('https://packages.confluent.io', true) %}
-{% set CLIENT_REPO_URL = lookup('env', 'CLIENT_REPO_URL') | default('https://packages.confluent.io', true) %}
